tutorials/timefreq/wavelets_tutorial.py

```diff
@@ -0,0 +1,166 @@
+"""
+Wavelets
+===============
+
+Perform time-frequency decomposition using wavelets.
+
+This tutorial will follow the ``neurodsp.timefrequency.wavelets`` module.
+It uses Morlet wavelets to transform time series data to a time-frequency
+representation of the data.
+"""
+
+###################################################################################################
+
+# Import neccessary functions and packages
+import numpy as np
+from scipy.signal import morlet
+
+from neurodsp.utils.data import create_freqs
+from neurodsp.utils.checks import check_n_cycles
+from neurodsp.utils.decorators import multidim
+
+# Import simulation code to create test Data
+from neurodsp.sim import sim_combined
+from neurodsp.utils import set_random_seed, create_times
+
+# Import utilities for loading and plotting data
+from neurodsp.utils.download import load_ndsp_data
+from neurodsp.plts.time_series import plot_time_series
+
+# Import functions for morlet Wavelets
+from neurodsp.timefrequency.wavelets import compute_wavelet_transform, convolve_wavelet
+
+###################################################################################################
+
+# Set the random seed, for consistency simulating data
+set_random_seed(0)
+
+###################################################################################################
+# Simulate time-frequency series data
+# -----------------------------
+#
+# First, we'll simulate the time frequency series using the function
+# 'sim_combined', which combines multiple component signals to simulate a signal.
+# It takes input parameter n_seconds and fs, which specify the simulation time and signal
+# sampling rate, respectively.
+# It also takes components which include 'sim_powerlaw',
+# 'sim_oscillation', and 'freq'.
+# These components must be specified separately.
+# The 'sim_powerlaw' function can be found under neurodsp.aperiodic.py.
+# It simulates a power law time series with a specified exponent.
+# The 'sim_oscillation' function can be found under neurodsp.periodic.py.
+# It simulates an oscillation, and takes input parameters n_seconds, fs, freq,
+# cycle, phase, and cycle_params.
+# The cycle is specified as sine, and the phase default is 0.
+# The cycle_params are the parameters for the simulated oscillation cycle,
+# which are already stored.
+#
+# We'll also need the function 'create_times' to create an array of time indices.
+# It takes input parameters n_seconds, fs, and start_val.
+# The n_seconds parameter is the length of the signal.
+# The fs parameter is the sampling frequency.
+# The start_val parameter is the starting value for the time definition, and it
+# has a default value of 0.
+
+###################################################################################################
+
+# Simulation settings
+fs = 500
+n_seconds = 10
+
+# Define simulation components
+components = {'sim_powerlaw': {exponent=-2.0, f_range=None, **filter_kwargs}, 'sim_oscillation' : {'freq': 10}})
```

Comment:
Also it looks like you have a hanging parentheses at the end and you can either define a dictionary like this `components=dict(sim_powerlaw=dict(exponent=-2))` or like this `components={'sim_powerlaw': {'exponent': -2}}`. I prefer the first for user-facing code such as a tutorial but it should be consistent with neurodsp style so check to see what other tutorials use. You're mixing the two with `{exponent=-2}` which doesn't work